File: backend/app/models/task.py
"""Task database model."""
from datetime import datetime
from typing import Optional
from sqlalchemy import Column, Integer, String, Boolean, DateTime, Text
from sqlalchemy.ext.declarative import declarative_base

Base = declarative_base()

class Task(Base):
    """Task model for database storage."""
    
    __tablename__ = "tasks"
    
    id: int = Column(Integer, primary_key=True, index=True)
    title: str = Column(String(255), nullable=False)
    description: Optional[str] = Column(Text, nullable=True)
    completed: bool = Column(Boolean, default=False, nullable=False)
    due_date: Optional[datetime] = Column(DateTime, nullable=True)
    priority: Optional[str] = Column(String(50), nullable=True)
    created_at: datetime = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at: datetime = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)
    
    def __repr__(self) -> str:
        """String representation of Task."""
        return f"<Task(id={self.id}, title='{self.title}', completed={self.completed})>"

File: frontend/src/types/task.ts
/**
 * Task type definitions
 */

export interface Task {
    id: number;
    title: string;
    description?: string;
    completed: boolean;
    due_date?: string;
    priority?: string;
    created_at: string;
    updated_at: string;
  }
  
  export interface TaskCreate {
    title: string;
    description?: string;
    due_date?: string;
    priority?: string;
  }
  
  export interface TaskUpdate {
    title?: string;
    description?: string;
    completed?: boolean;
    due_date?: string;
    priority?: string;
  }
  
  export interface NaturalLanguageRequest {
    text: string;
  }